Src/Plugins/Security/SEC101_054.AlibabaAccessKeyValidator.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+// From the Nuget package
+using Aliyun.Acs.Core;
+using Aliyun.Acs.Core.Exceptions;
+using Aliyun.Acs.Core.Profile;
+
+// From the DLL
+using Aliyun.Acs.Iot.Model.V20170420;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security.Utilities;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    public class AlibabaAccessKeyValidator : ValidatorBase
+    {
+        internal static AlibabaAccessKeyValidator Instance;
+
+        protected const string KeyNotFound = "InvalidAccessKeyId.NotFound";
+        protected const string InvalidSecret = "SDK.InvalidAccessKeySecret";
+
+        static AlibabaAccessKeyValidator()
+        {
+            Instance = new AlibabaAccessKeyValidator();
+        }
+
+        public static string IsValidStatic(ref string matchedPattern,
+                                           ref Dictionary<string, string> groups,
+                                           ref string failureLevel,
+                                           ref string fingerprint,
+                                           ref string message)
+        {
+            return IsValidStatic(Instance,
+                                 ref matchedPattern,
+                                 ref groups,
+                                 ref failureLevel,
+                                 ref fingerprint,
+                                 ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return IsValidDynamic(Instance,
+                                   ref fingerprint,
+                                   ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            if (!groups.TryGetNonEmptyValue("account", out string account) ||
+                !groups.TryGetNonEmptyValue("password", out string password))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Account = account,
+                Password = password,
+            }.ToString();
+
+            return nameof(ValidationState.AuthorizedWarning);
```

Comment:
>AuthorizedWarning [](start = 42, length = 17)

use unknown

Reply:
Done.

---
In reply to: [591816424](https://github.com/microsoft/sarif-pattern-matcher/pull/285#discussion_r591816424) [](ancestors = 591816424)